api4/team.go

```diff
@@ -605,7 +625,7 @@ func searchTeams(c *Context, w http.ResponseWriter, r *http.Request) {
 	if c.App.SessionHasPermissionTo(c.App.Session, model.PERMISSION_MANAGE_SYSTEM) {
 		teams, err = c.App.SearchAllTeams(props.Term)
 	} else {
-		teams, err = c.App.SearchOpenTeams(props.Term)
+		teams, err = c.App.SearchPublicTeams(props.Term)
```

Comment:
Should this actually be governed by `PERMISSION_LIST_PUBLIC_TEAMS` and `PERMISSION_LIST_PRIVATE_TEAMS` too, rather than `PERMISSION_MANAGE_SYSTEM`? Or is that incorrect, or is it just not necessary?